### ISSUE TITLE
Wrong warning on sigils

### ISSUE THREAD

**Issue by eksperimental** [rrrene/credo#813]
### Environment

* Credo version (`mix credo -v`): 1.5.0-rc.2
* Erlang/Elixir version (`elixir -v`):  Elixir 1.11.0-rc.0 (48cee5a) (compiled with Erlang/OTP 23)
* Operating system: Linux

### Actual outcome

>   Code Readability                                                                                                                                                       
>  [R] ↗ Function/macro/guard names should be written in snake_case.
>        lib/ord_map.ex:311:12 #(OrdMap.sigil_O)
> 

It is a macro sigil, and it is giving this error. I have checked the Credo code and it is not supposed to warn when it is a sigil, but it does.

**Comment by rrrene** [comment]
> and it is not supposed to warn when it is a sigil

Not only that, but the test - coincidentally? - also uses `sigil_O` as the test subject.

I could not reproduce this with Elixir 1.11.0-rc.0 on either Erlang/OTP 21 or 23.

Can you provide any additional detail? Is this a straight vanilla macro or is some meta-programming involved? What does the function/macro signature look like?

**Comment by eksperimental** [comment]
I narrowed it down and it happens when there is a guard in the macro.

```elixir
defmodule CredoBug do
  @moduledoc false
  defmacro sigil_O(arg, modifiers) when is_tuple(arg) do
    :foo
  end
end
```

remove the guard and Credo won't complain

**Comment by eksperimental** [comment]
> I narrowed it down and it happens when there is a guard in the macro.
> 
> ```elixir
> defmodule CredoBug do
>   @moduledoc false
>   defmacro sigil_O(arg, modifiers) when is_tuple(arg) do
>     :foo
>   end
> end
> ```
> 
> remove the guard and Credo won't complain

it happens when same with a guard in a function,.

**Comment by eksperimental** [comment]
Looking at the code, I realized there is no reason for give a special treatment to sigils, because they comply with the snake_case rule. 

I realized that you are excluding public sigils only, but sigils could be private as well.

I think the rule needs to be revamped, because we could have  a name such `defp clean_HTTP_url(string)` which uses snake case but it will be caught as an issue.

**Comment by eksperimental** [comment]
Maybe another rule where no upper case should be used in guards, functions and macros if you want to discourage users from using `clean_HTTP_url` and use `clean_http_url` instead

**Comment by rrrene** [comment]
Thx for reporting this! :+1: It should be be fixed in `1.5.0-rc.5`.

**Comment by eksperimental** [comment]
Awesome, thank you
